source/ddbus/thin.d

```diff
@@ -264,6 +267,66 @@ struct DBusAny {
       ~ ", " ~ valueStr ~ ")";
   }
 
+  /++
+    Returns the value stored in the DBusAny object by specifying the type.
+
+    Parameters:
+      T = The requested type. The currently stored value must match the
+        requested type exactly.
+
+    Throws:
+      TypeMismatchException if the DBus type of the current value of the
+      DBusAny object is not the same as the DBus type used to represent T.
+  +/
+  T get(T)() @property const
+    if (staticIndexOf!(T, BasicTypes) >= 0)
+  {
+    enforce(type == typeCode!T,
+      new TypeMismatchException(
+        "Cannot get a " ~ T.stringof ~ " from a DBusAny with"
+          ~ " a value of DBus type '" ~ typeSig ~ "'.", typeCode!T, type));
+
+    static if (isIntegral!T) {
+      enum memberName =
+        (isUnsigned!T ? "uint" : "int") ~ (T.sizeof << 3).to!string;
+      return __traits(getMember, this, memberName);
+    } else static if (is(T == double)) {
+      return float64;
+    } else static if (is(T == string)) {
+      return str;
+    } else static if (is(T == ObjectPath)) {
+      return obj;
+    } else static if (is(T == bool)) {
+      return boolean;
+    }
```

Comment:
```d
else static assert(false, "Can't get " ~ T.stringof ~ " from a DBusAny, can only get integral types, double, string, ObjectPath, bool, DBusAny[] and ubyte[]");
```

Reply:
That assert would never trigger because the static ifs list all types allowed by the constraint. Still it makes some sense to add it, though with a different message, just because reaching the `else` would indicate a ddbus bug.

Reply:
oh yeah you are right, didn't pay attention to that, just saw an unterminated `else static if` block. But I think there should be a static assert false without error message in case we add a new type to BasicTypes and forget to update this (like SIGNATURE or UNIX_FD)